include/deal.II/base/mpi.h

```diff
@@ -1538,6 +1539,85 @@ namespace Utilities
 #  endif
     }
 
+    namespace internal
+    {
+      template <typename T>
+      ArrayView<T>
+      builtin_broadcast(const MPI_Comm &    comm,
+                        const ArrayView<T> &object_to_send,
+                        const unsigned int  root_process)
+      {
+#  ifndef DEAL_II_WITH_MPI
+        (void)comm;
+        (void)root_process;
+        return object_to_send;
+#  else
+        MPI_Datatype datatype =
+          dealii::Utilities::MPI::internal::mpi_type_id(object_to_send);
+
+        ArrayView<T> buffer;
+        if (this_mpi_process(comm) == root_process)
+          {
+            buffer = object_to_send;
```

Comment:
This isn't going to work. On anything other than the root process, `buffer` is equivalent to a null pointer. You're asking `MPI_Bcast` to put content into it, but there is no "there" to put stuff in.

You will have to allocate the memory on the calling side of this function and provide it to the function via an argument.

Reply:
This was actually a placeholder since I was still waiting to know how to communicate the size of the array between the processes, I should have mentioned that in a comment. In 147b71fd64585107f0780dbb0826814473a6d371 I removed the function as suggested below.